### PR TITLE
net/ndproxy: Add ndproxy manual

### PR DESCRIPTION
**Pull request by Monviech**
For https://github.com/opnsense/plugins/pull/4348

### CODE REVIEW

File: source/manual/ndproxy.rst [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+--------------------------------------------------
+
+Install ``os-ndproxy`` from :menuselection:`System --> Firmware --> Plugins`.
+
+
+Important configuration details
+--------------------------------------------------
+
+- **Promiscuous Mode**:
+    The listening interface (WAN) must be set to promiscuous mode.
```
Comment (fichtner) [inline source/manual/ndproxy.rst:36]: If the WAN is a VLAN you need to assign a parent and set promisc mode there. May be worth a note?

File: source/manual/ndproxy.rst [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+
+Important configuration details
+--------------------------------------------------
+
+- **Promiscuous Mode**:
+    The listening interface (WAN) must be set to promiscuous mode. 
+    Otherwise the router can not join multicast groups to respond to solicitations for hosts in the LAN.
+  
+- **Link-Local Address**:
+    The listening interface (WAN) requires at least a link-local IPv6 address.
```
Comment (fichtner) [inline source/manual/ndproxy.rst:40]: This requirement is true but in 99% of cases that is a given (unless somebody wedges WAN on a bridge where link-local option was not set -- but the text here will not help those either)

File: source/manual/ndproxy.rst [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+    If you want to add a GUA (Global Unicast Address) to WAN, it must be with /128 Prefix.
```
Comment (fichtner) [inline source/manual/ndproxy.rst:41]: Only if it is in the same /64 being proxied.

File: source/manual/ndproxy.rst [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+  
+.. Attention::
+
+   If WAN and LAN both have a GUA configured with the same /64 Prefix, this setup will not work. The network stack will not be able
+   to decide the sending interface, and routing will fail.
+   
+- **Switch configuration**:
+
+   If there is a switch between the PE and CPE router, ensure there is no MLD snooping or Multicast (IGMP) snooping configured
```
Comment (fichtner) [inline source/manual/ndproxy.rst:50]: 👍

File: source/manual/ndproxy.rst [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+        |                 | LAN                          WAN |                 |
+        |    Router PE    |----------------------------------|    Router CPE   |
+        |                 | fe80::1/64            fe80::2/64 |    (ndproxy)    |
+        +-----------------+ 2001:db8::1/64   2001:db8::2/128 +-----------------+
+                | WAN                                             LAN |
+                |                                                     |
+        Prefix Delegation: /56                                        |
+                |                                                     |
+          fe80::1/64                                            fe80::1/64
+          2001:db8::/56                                         2001:db8::3/64
```
Comment (fichtner) [inline source/manual/ndproxy.rst:78]: Not sure if 2001:db8::3/64 is strictly necessary, especially with a shifting prefix this is impractical. But in the dynamic case a track interface setting works anyway?

File: source/manual/ndproxy.rst [hunk 6]
```diff
@@ -0,0 +1,229 @@
...
+**Promiscous Mode**                             ``X`` 
+                                                (important to respond to all NDP Multicasts)
+**Prefix Delegation Size**                      ``64``
+**Request Prefix Only**                         ``X``
+                                                (optional)
+==============================================  ====================================================================
+
+.. Note::
+
+   `Request Prefix Only` is optional, but needed if your WAN interface would autoconfigure a /64 GUA.
```
Comment (fichtner) [inline source/manual/ndproxy.rst:153]: Don't understand this. SLAAC is a separate range and /64. IA-NA is /128 -- IA-PD is /64 but not directly assigned to WAN so LAN could already use it?

**Review by fichtner — COMMENTED** [review]
Maybe we should discuss the use cases before continuing the review. Looks like I'm missing something so better to clear that up first :)

**Comment by Monviech** [comment]
> Maybe we should discuss the use cases before continuing the review. Looks like I'm missing something so better to clear that up first :)

Sounds like a good plan, I appreciate the help. :)

File: source/manual/ndproxy.rst [hunk 7]
```diff
@@ -0,0 +1,229 @@
...
+**IPv6 Configuration Type**                     ``Static IPv6``
+**IPv6 address**                                ``2001:db8::1/64``
+==============================================  ====================================================================
+
+Go to :menuselection:`Services --> ISC DHCPv6 --> LAN`
+
+==============================================  ====================================================================
+**Enable**                                      ``X``
+**Range**                                       from: ``2001:db8::2`` to: ``2001:db8::2``
+**Prefix Delegation Range**                     from: ``2001:db8::`` to: ``2001:db8::``
```
[the remark this answers is not in the document]
Comment (Monviech) [reply source/manual/ndproxy.rst:118]: The GUA is not strictly needed for DHCPv6 to work here. The range can also be left empty.

But getting a /128 on the WAN can be important, e.g. for IPv6 VPN tunnels.

File: source/manual/ndproxy.rst (hunk 6 above)
Comment (Monviech) [reply source/manual/ndproxy.rst:153]: Im not sure here either. It works with this enabled, and also disabled.

But if Router Advertisements would be set to Assisted on the Upstream Router, there could be a DHCPv6 /128 and a SLAAC /64 configured at the same time. I saw that happen, maybe it was a weird one time thing on my side but it wrecked my routing until I found that out.

**Review by fichtner — APPROVED** [review]
great, thanks for the redesign ❤️

**Comment by Monviech** [comment]
Come on lets gooooooo

**Pull request merged**